fennel/client_tests/test_discrete_window_aggregation.py

```diff
@@ -0,0 +1,190 @@
+from datetime import datetime, timezone, timedelta
+
+import pandas as pd
+
+from fennel.connectors import Webhook, source
+from fennel.datasets import Count, Dataset, dataset, field, pipeline, Sum
+from fennel.dtypes import Tumbling, Session, Hopping
+from fennel.lib import inputs
+from fennel.testing import mock
+
+__owner__ = "nitin@fennel.ai"
+webhook = Webhook(name="fennel_webhook")
+
+
+@mock
+def test_discrete_hopping_tumbling_window_aggregation(client):
+    @source(webhook.endpoint("Transactions"), cdc="append", disorder="14d")
+    @dataset
+    class Transactions:
+        id: str
+        user_id: str
+        amount: float
+        t: datetime
+
+    @dataset(index=True)
+    class Stats:
+        user_id: str = field(key=True)
+        sum: float
+        count: int
+        t: datetime
+
+        @pipeline
+        @inputs(Transactions)
+        def pipeline(cls, event: Dataset):
+            return (
+                event.dedup("id")
+                .groupby("user_id")
+                .aggregate(
+                    count=Count(window=Tumbling("1d")),
+                    sum=Sum(window=Tumbling("1d"), of="amount"),
+                )
+            )
+
+    client.commit(datasets=[Transactions, Stats], message="first_commit")
+
+    now = datetime.now(timezone.utc)
+    now_1d = datetime.now(timezone.utc) - timedelta(days=1)
+    now_2d = datetime.now(timezone.utc) - timedelta(days=2)
+    now_3d = datetime.now(timezone.utc) - timedelta(days=3)
+
+    df = pd.DataFrame(
+        {
+            "id": [1, 2, 3, 4, 5, 6, 7, 8, 9, 10],
+            "user_id": [1, 1, 1, 1, 1, 2, 2, 2, 3, 3],
+            "amount": [100, 200, 300, 400, 500, 600, 700, 800, 900, 1000],
+            "t": [
+                now,
+                now_1d,
+                now_1d,
+                now_2d,
+                now_1d,
+                now_1d,
+                now_2d,
+                now_3d,
+                now_3d,
+                now_3d,
+            ],
+        }
+    )
+    client.log("fennel_webhook", "Transactions", df)
+    client.sleep()
```

Comment:
lets add the sleep when we convert this into a integration test

Reply:
the sleep in normal client is zero